src/nvim/mark_defs.h

```diff
@@ -25,24 +53,39 @@
 /// but they are not saved in ShaDa files.
 #define NLOCALMARKS     (NMARKS + 3)
 
+/// Max value of local mark
+#define NMARK_LOCAL_MAX 126  // Index of '~'
+
 /// Maximum number of marks in jump list
 #define JUMPLISTSIZE    100
 
 /// Maximum number of tags in tag stack
 #define TAGSTACKSIZE    20
 
+/// Represetns view in which the mark was created
+typedef struct fmarkv {
+  linenr_T topline_offset;  // Amount of lines from the mark lnum to the top of the window.
+} fmarkv_T;
+
+#define INIT_FMARKV { 0 }
+
 /// Structure defining single local mark
 typedef struct filemark {
-  pos_T mark;           ///< Cursor position.
```

Comment:
I went by the [dev_style](https://github.com/muniter/neovim/blob/825727875b081a9ace8dce251b9536bee7eebe3d/runtime/doc/dev_style.txt#L551) examples. Since I added something I went for the same style, I'm happy to revert.